File: Aksesoris.java
public class Aksesoris extends Petshop {
    private String jenis;
    private String bahan;
    private String warna;

    public Aksesoris() {
        super();
        this.jenis = "";
        this.bahan = "";
        this.warna = "";
    }

    public Aksesoris(int id, String namaProduk, int hargaProduk, int stokProduk, String jenis, String bahan, String warna) {
        super(id, namaProduk, hargaProduk, stokProduk);
        this.jenis = jenis;
        this.bahan = bahan;
        this.warna = warna;
    }

    public String getJenis() {
        return jenis;
    }

    public void setJenis(String jenis) {
        this.jenis = jenis;
    }

    public String getBahan() {
        return bahan;
    }

    public void setBahan(String bahan) {
        this.bahan = bahan;
    }

    public String getWarna() {
        return warna;
    }

    public void setWarna(String warna) {
        this.warna = warna;
    }
}

File: Aksesoris.cpp
#include <iostream>
#include <list>
#include <string>
#include "Petshop.cpp"

using namespace std;
    
class Aksesoris : public Petshop {
    private:
        string jenis;
        string bahan;
        string warna;

    public:
        Aksesoris() {
            jenis = "";
            bahan = "";
            warna = "";
        }

        Aksesoris(int id, string nama_produk, int harga_produk, int stok_produk, string jenis, string bahan, string warna) : Petshop(id, nama_produk, harga_produk, stok_produk) {
            this->jenis = jenis;
            this->bahan = bahan;
            this->warna = warna;
        }

        void setJenis(string jenis) {
            this->jenis = jenis;
        }

        string getJenis() {
            return jenis;
        }

        void setBahan(string bahan) {
            this->bahan = bahan;
        }

        string getBahan() {
            return bahan;
        }

        void setWarna(string warna) {
            this->warna = warna;
        }

        string getWarna() {
            return warna;
        }

        int jenisLength() {
            return jenis.length();
        }

        int bahanLength() {
            return bahan.length();
        }

        int warnaLength() {
            return warna.length();
        }
};